### PR TITLE
Update Makefile, README and CHANGELOG

### PR DESCRIPTION
**Pull request by xthexder**
This should put us at a release-able 2.0rc2 state.

Things I did:
- Split the changelog into an rc1 and rc2 version
- Updated the Makefile for `toxiproxy-cli` and `toxiproxy-server`
- Replaced the Curl example in the README with a CLI example
- Tiny cli help text changes

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -60,7 +63,7 @@ $(DEB): tmp/build/toxiproxy-server-linux-amd64 tmp/build/toxiproxy-client-linux-
 		--url "https://github.com/Shopify/toxiproxy" \
 		$(word 1,$^)=/usr/bin/toxiproxy \
 		$(word 1,$^)=/usr/bin/toxiproxy-server \
-		$(word 2,$^)=/usr/bin/toxiproxy-client \
```
Comment (jpittis) [inline Makefile]: Should these be $(CLI_NAME) / $(SERVER_NAME)?

Comment (xthexder) [inline Makefile]: whoops, yes it should

File: Makefile [hunk 2]
```diff
@@ -59,8 +62,8 @@ $(DEB): tmp/build/toxiproxy-server-linux-amd64 tmp/build/toxiproxy-client-linux-
 		--description "TCP proxy to simulate network and system conditions" \
 		--url "https://github.com/Shopify/toxiproxy" \
 		$(word 1,$^)=/usr/bin/toxiproxy \
```
Comment (xthexder) [inline Makefile]: Not sure if we should leave this or not, I don't really have much of an opinion on it

**Comment by jpittis** [comment]
Walked through the cli example! :smile:
Besides silly nitpick, LGTM.

File: README.md [hunk 3]
```diff
@@ -58,7 +58,7 @@ stopping you from creating a client in any other language (see
   1. [Proxy fields](#proxy-fields)
   2. [Toxic fields](#toxic-fields)
   3. [Endpoints](#endpoints)
-  4. [Curl example](#curl-example)
+  4. [CLI example](#cli-example)
```
Comment (Sirupsen) [inline README.md]: This should probably not be under `HTTP API` anymore.

**Comment by Sirupsen** [comment]
LGTM

**Pull request merged**